AlmaLinux 8 build fails with "invalid application of 'sizeof' to incomplete type 'p_text_poke_loc'"

This just started happening today, only noticed in forks of this repo so far:

https://github.com/solardiz/lkrg/actions/runs/3330536896/jobs/5509151190

```
In file included from /__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/../../../../modules/database/p_database.h:87,
                 from /__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/../../../../p_lkrg_main.h:399,
                 from /__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c:31:
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/../../../../modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.h:82:42: error: invalid application of 'sizeof' to incomplete type 'p_text_poke_loc' {aka 'struct text_poke_loc'}
 #define P_TP_VEC_MAX (PAGE_SIZE / sizeof(p_text_poke_loc))
                                          ^~~~~~~~~~~~~~~
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c:35:38: note: in expansion of macro 'P_TP_VEC_MAX'
 static unsigned long p_jl_batch_addr[P_TP_VEC_MAX];
                                      ^~~~~~~~~~~~
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c: In function 'p_arch_jump_label_transform_apply_entry':
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c:74:72: error: invalid application of 'sizeof' to incomplete type 'p_text_poke_loc' {aka 'struct text_poke_loc'}
       p_tmp = (p_text_poke_loc *)&P_SYM(p_tp_vec)[p_jl_batch_nr*sizeof(p_text_poke_loc)];
                                                                        ^~~~~~~~~~~~~~~
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c:88:18: error: dereferencing pointer to incomplete type 'p_text_poke_loc' {aka 'struct text_poke_loc'}
       if ( (p_tmp->len == 5
                  ^~
At top level:
/__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c:35:22: warning: 'p_jl_batch_addr' defined but not used [-Wunused-variable]
 static unsigned long p_jl_batch_addr[P_TP_VEC_MAX];
                      ^~~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:317: /__w/lkrg/lkrg/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:1584: _module_/__w/lkrg/lkrg] Error 2
make[1]: Leaving directory '/usr/src/kernels/4.18.0-372.32.1.el8_6.x86_64'
```

My guess is it's some RHEL8 change that we now need to adapt to, but I didn't look into this for real yet.

--- a/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c
+++ b/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c
@@ -72,14 +72,17 @@ notrace int p_arch_jump_label_transform_apply_entry(struct kretprobe_instance *p
 
    for (p_jl_batch_nr = 0; p_cnt < p_nr; p_cnt++) {
       p_tmp = (p_text_poke_loc *)&P_SYM(p_tp_vec)[p_jl_batch_nr*sizeof(p_text_poke_loc)];
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0) || \
+    defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
       if ( (p_tmp->opcode == CALL_INSN_OPCODE
             || p_tmp->opcode == JMP32_INSN_OPCODE
             || p_tmp->opcode == INT3_INSN_OPCODE
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) || \
+    defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
             || p_tmp->opcode == RET_INSN_OPCODE
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0) || \
+    defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
             || p_tmp->opcode == JMP8_INSN_OPCODE
 #endif
             ) &&
@@ -101,7 +104,8 @@ notrace int p_arch_jump_label_transform_apply_entry(struct kretprobe_instance *p
 
 #endif
          p_jl_batch_addr[p_jl_batch_nr++] =
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0) || \
+    defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
                   (unsigned long)p_tmp->rel_addr +
                   (unsigned long)p_db.kernel_stext.p_addr;
 #else

--- a/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.h
+++ b/src/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.h
@@ -38,6 +38,10 @@
 
 #include <asm/linkage.h> /* for ASM_RET */
 
+#if defined(RHEL_RELEASE_CODE) && defined(DISP32_SIZE)
+ #define P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL 1
+#endif
+
 /*
  * This can be extended to other LTS or active branches if and when they
  * receive the variable length JUMP_LABEL feature backport, although the
@@ -55,8 +59,10 @@
 
 #include <asm/text-patching.h>
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
- #if !P_LKRG_KERNEL_HAS_VAR_LEN_JUMP_LABEL
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0) || \
+    defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
+ #if !P_LKRG_KERNEL_HAS_VAR_LEN_JUMP_LABEL || \
+     defined(P_LKRG_KERNEL_RHEL_VAR_LEN_JUMP_LABEL)
 typedef struct _p_text_poke_loc {
     s32 rel_addr; /* addr := _stext + rel_addr */
     s32 rel32;

